core/includes/diff.inc

```diff
@@ -982,11 +1115,13 @@ abstract class DiffFormatter {
   /**
    * Format a diff.
    *
-   * @param $diff Diff A Diff object.
+   * @param Diff $diff
+   *   A Diff object.
    *
-   * @return string The formatted output.
+   * @return string
+   *   The formatted output.
    */
-  public function format($diff) {
+  public function format(Diff $diff) {
```

Comment:
In this function it's clear that it can't be anything else than "Diff", so this type hint seems OK. 😉